tests/suites/test_suite_cipher.aes.data

```diff
@@ -2111,99 +2111,99 @@ depends_on:MBEDTLS_USE_PSA_CRYPTO:MBEDTLS_TEST_DEPRECATED:MBEDTLS_AES_C:MBEDTLS_
 test_vec_crypt:MBEDTLS_CIPHER_AES_256_CBC:MBEDTLS_DECRYPT:"0000000000000000000000000000000000000000000000000000000000000000":"00000000000000000000000000000000":"5c9d844ed46f9885085e5d6a4f94c7d7":"014730f80ac625fe84f026c60bfd547d":0:1
 
 AES-128-ECB crypt Encrypt NIST KAT #1 PSA
-depends_on:MBEDTLS_USE_PSA_CRYPTO:MBEDTLS_TEST_DEPRECATED:MBEDTLS_AES_C
+depends_on:MBEDTLS_USE_PSA_CRYPTO:MBEDTLS_TEST_DEPRECATED:MBEDTLS_AES_C:PSA_WANT_ALG_ECB_NO_PADDING
```

Comment:
Shouldn't `PSA_WANT_ALG_ECB_NO_PADDING` be instead of `MBEDTLS_AES_C`, rather than in addition?

Reply:
`PSA_WANT_ALG_ECB_NO_PADDING` might still be defined when there's no `MBEDTLS_AES_C`, and underneath `mbedtls_cipher_crypt` will use (after more or less layers, depending on whether the driver is used or not) `mbedtls_aes_xxx` anyway... Am I missing something here?

Reply:
test_vec_crypt calls `mbedtls_cipher_setup_psa`, which sets up a PSA cipher context (`ctx->psa_enabled = 1`). Then `mbedtls_cipher_setkey` and `mbedtls_cipher_crypt` both see that `ctx->psa_enabled == 1` and so call PSA functions. I don't see where this would fail if AES is implemented only as a PSA driver.

Reply:
OK, so if we fully support doing this via the driver, the dependency should be `MBEDTLS_HAS_ALG_AES_VIA_CIPHER_OR_PSA`. This define will be added here: https://github.com/Mbed-TLS/mbedtls/pull/6462
Is the support for cipher accelerators in `libtestdriver1` and dispatching via `psa_crypto_driver_wrappers` done already (I can see that it's possible to have MBEDTLS_PSA_ACCEL_KEY_TYPE_AES` already in `config_psa.h)?
If not, I would assume that the library always lands in the path without the driver, so AES is needed, and, as a part of  https://github.com/orgs/Mbed-TLS/projects/1#column-19075367, this will be changed to the `HAS_ALG_AES_VIA_CIPHER_OR_PSA` family of defines.

Reply:
As noted in chat, actually
```suggestion
depends_on:MBEDTLS_USE_PSA_CRYPTO:MBEDTLS_TEST_DEPRECATED:PSA_WANT_KEY_TYPE_AES:PSA_WANT_ALG_ECB_NO_PADDING
```

Reply:
Fixed for tests that were in the basic scope of this PR as well as other that I found from ccm and gcm. Do you think @gilles-peskine-arm that it would be helpful to open an issue about finding a way to test to discover such cases of mistaken dependencies?

Reply:
I'm not sure if we need a systematic campaign, or that's just something we'll gradually improve when we improve testing with drivers. We'd like to retire the cipher module in the next major release, and when we do so we'll have fewer difficulties of this kind (hashes are easier because they map 1-to-1 between legacy and PSA, and pk are easier because there are fewer algorithms to enumerate). Can we wait it out?

If anyone, that's a question for @mpg who's done the planning of the migration to USE_PSA. Do we need more effort on testing test case dependencies in legacy APIs, especially cipher?

Reply:
I'd say it depends on the scope of the question: if it's about tests for `cipher_setup_psa()` I don't think we need a systematic campaign to look for missing dependencies, as `mbedtls_cipher_setup_psa()` is already officially deprecated, in particular not used anywhere in the library, so unlikely to cause issues elsewhere, and if it does we can fix problems when they arise.

If the scope is larger, like the whole cipher module, well obviously I think we needed some level of systematic hunting fro dependency issues, which is what we revived `depends.py`. Also, I expect the upcoming work where we compare tests skipped in a driver-based build vs the corresponding reference build will catch some issues (I think it would have caught the `AES_C` -> `WANT_KEY_TYPE_AES` part here, but not the `ALG_ECB_NO_PADDING` one). Overall I'd say that should be enough and we don't need to be beyond that in a systematic way - any other issue can and should be fixed as and when they're found.